### PR TITLE
Added more cheats

### PR DESCRIPTION
**Pull request by AdamPlenty**
Added cheats for levelling creatures up, killing players, editing Dungeon Heart health, and Querying creatures not under your control.

### CODE REVIEW

File: src/magic.c [hunk 1]
```diff
@@ -91,6 +92,13 @@ TbBool can_cast_spell_f(PlayerNumber plyr_idx, PowerKind pwkind, MapSubtlCoord s
...
         return false;
     }
+    struct PlayerInfo* player = get_player(plyr_idx);
+    if (player->work_state == PSt_FreeCtrlDirect)
+    {
+        return true;
+    }
+    else
+    {
     TbBool cast_at_xy;
```
Comment (Loobinex) [inline src/magic.c:102]: Could you indent the entire else statement, to the level of "return true" above. So 4 spaces in front lines 102-167.

File: src/packets.c [hunk 2]
```diff
@@ -946,7 +946,7 @@ TbBool process_dungeon_control_packet_clicks(long plyr_idx)
         if (player->work_state == PSt_CtrlDirect)
             thing = get_creature_near_for_controlling(plyr_idx, x, y);
         else
-            thing = get_creature_near_for_controlling(plyr_idx, x, y); // TODO Change to allow controlling any owner
+            thing = get_creature_near(x, y); // TODO Change to allow controlling any owner
```
Comment (Loobinex) [inline src/packets.c:949]: Didn't you already manage to implement this TODO statement? If so, the comment can be removed.

File: src/packets.c [hunk 3]
```diff
@@ -1223,30 +1223,38 @@ TbBool process_dungeon_control_packet_clicks(long plyr_idx)
...
-                    struct Room* room = room_get(slb->room_index);
+                    room = room_get(slb->room_index);
                     destroy_room_leaving_unclaimed_ground(room);
                 }
             unset_packet_control(pckt, PCtr_LBtnRelease);
         }
         break;
     case PSt_KillCreatr:
+            thing = get_creature_near(x, y);
+            if (!thing_is_creature(thing))
```
Comment (Loobinex) [inline src/packets.c:1234]: The indentation seems wrong here, the 'if' is indented more than the 'else' statement.

File: src/packets.c [hunk 4]
```diff
@@ -1223,30 +1223,38 @@ TbBool process_dungeon_control_packet_clicks(long plyr_idx)
...
+             if (player->thing_under_hand > 0)
             {
                 kill_creature(thing, INVALID_THING, -1, CrDed_NoUnconscious);
             }
         unset_packet_control(pckt, PCtr_LBtnRelease);    
         }
         break;
     case PSt_ConvertCreatr:
+            thing = get_creature_near(x, y);
+            if (!thing_is_creature(thing))
```
Comment (Loobinex) [inline src/packets.c:1249]: Here the 'if' line should also be indented 4 spaces less.

File: src/packets.c [hunk 5]
```diff
@@ -1281,11 +1289,226 @@ TbBool process_dungeon_control_packet_clicks(long plyr_idx)
...
               }
-              place_slab_type_on_map(slbkind, slab_subtile(subtile_slab_fast(stl_x), 0), slab_subtile(subtile_slab_fast(stl_y), 0), plyr_idx, 0);
+              place_slab_type_on_map(slbkind, slab_subtile(subtile_slab_fast(stl_x), 0), slab_subtile(subtile_slab_fast(stl_y), 0), get_selected_player_for_cheat(plyr_idx), 0);
+          }
+            unset_packet_control(pckt, PCtr_LBtnRelease);
+        }
+        break;
+    case PSt_LevelCreatureUp:
+            thing = get_creature_near(x, y);
+            if (!thing_is_creature(thing))
```
Comment (Loobinex) [inline src/packets.c:1299]: indentation,....

File: src/packets.c [hunk 6]
```diff
@@ -1281,11 +1289,226 @@ TbBool process_dungeon_control_packet_clicks(long plyr_idx)
...
+            player->thing_under_hand = 0;
+        else
+            player->thing_under_hand = thing->index;
+        if (((pckt->control_flags & PCtr_LBtnRelease) != 0) && ((pckt->control_flags & PCtr_MapCoordsValid) != 0))
+        {
+             if (player->thing_under_hand > 0)
+          {
```
Comment (Loobinex) [inline src/packets.c:1306]: These 3 lines can be indented more.

File: src/packets.c [hunk 7]
```diff
@@ -1281,11 +1289,226 @@ TbBool process_dungeon_control_packet_clicks(long plyr_idx)
...
+                if (thing_is_dungeon_heart(thing))
+                    {
+                        thing->health++;
+                    }
+            }
+            else if (is_key_pressed(KC_EQUALS, KMod_SHIFT))
+            {
+                if (thing_is_dungeon_heart(thing))
+                    {
+                        thing->health++;
```
Comment (Loobinex) [inline src/packets.c:1337]: I believe you want to remove health here right?

**Comment by Loobinex** [comment]
You forgot to recalculate the room efficiency if you placed a slab inside an existing room.

**Comment by AdamPlenty** [comment]
> 
> 
> You forgot to recalculate the room efficiency if you placed a slab inside an existing room.

I thought that automatically happened. How do I do that?

File: src/packets.c (hunk 7 above)
Comment (AdamPlenty) [reply src/packets.c:1340]: > 
> 
> I believe you want to remove health here right?

Remove health where? There are keys for adding health, and keys for removing health.

**Pull request merged**